### PR TITLE
MiscellaneousChanges

### PR DESCRIPTION
**Pull request by xinlian12**
Changes included:
- CI/Integration Test pipeline: https://github.com/Azure/azure-sdk-for-java/issues/38770
- Fixed 400 error when using sink connector for containers with nested partition key path
- clean few TODO comments

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

**Comment by xinlian12** [comment]
Manually triggered kafka pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3721107&view=results
Manually triggered cosmos tests pipeline: https://dev.azure.com/azure-sdk/internal/_build/results?buildId=3721206&view=results

File: sdk/cosmos/kafka-cosmos-matrix.json [hunk 1]
```diff
@@ -3,12 +3,11 @@
     "Cosmos": {
       "Session_Integration": {
         "ArmTemplateParameters": "@{ defaultConsistencyLevel = 'Session'; enableMultipleWriteLocations = $false }",
-        "ProfileFlag": "-P kafka-integration",
-        "Pool": "azsdk-pool-mms-ubuntu-2004-general",
-        "OSVmImage": "MMSUbuntu2004"
+        "ProfileFlag": "-Pkafka",
```
Comment (kushagraThapar) [inline sdk/cosmos/kafka-cosmos-matrix.json:6]: why this change?

File: sdk/cosmos/kafka-testcontainer-matrix.json [hunk 2]
```diff
@@ -0,0 +1,13 @@
+{
+  "matrix": {
```
Comment (kushagraThapar) [inline sdk/cosmos/kafka-testcontainer-matrix.json:2]: why do we have 2 matrix files?

**Review by kushagraThapar — COMMENTED** [review]
@xinlian12 can we please discuss these 2 comments?
